Enterprise-CMCS/security-hub-visibility
# Add action

This change adds the necessary source files and CI to create a GitHub action that syncs AWS Security Hub with Jira.  It's a thin wrapper around https://github.com/Enterprise-CMCS/macpro-security-hub-sync

The action is written in Typescript, and uses [ncc](https://github.com/vercel/ncc) to compile the module into a single file for distribution.  This file is referenced in the `action.yml` file that defines the action.

For CI, there are workflows that: 
- make sure that `ncc` was run
- run CodeQL
- test the action itself

Manual testing:
I ran this against MACPro's Jira with a test project that I created: https://qmacbis.atlassian.net/jira/software/projects/BTSHV/boards/254/backlog
and confirmed that I could:
- add issues to an epic or not
- add issues to a sprint or not (a custom field)
- filter different severities
- ignore different Jira statuses
- close an issue unresolved findings and see the issue recreated on the next run
- resolve a finding and see the issue closed on the next run
